cmsController.php

```diff
@@ -0,0 +1,195 @@
+<?php
+$db = new PDO('mysql:host=127.0.0.1; dbname=portfolioKyam', 'root');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+/**
+ * Gets about sections name, content from Db and turns into an array
+ *
+ * @param $db PDO to select from
+ * @return assoc array of about sections and content
+ */
+function FillAbout (PDO $db):array {
+    $query=$db->prepare("SELECT `name`,`content` FROM `staticContent` WHERE `name` != 'submitAbout' ORDER BY `name` DESC;");
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * Gets portfolio items names from DB and turns into an array
+ *
+ * @param $db PDO to select from
+ * @return assoc array of pf item names
+ */
+function portfolioList (PDO $db):array {
+    $query=$db->prepare("SELECT `title` FROM `portfolioItems` WHERE `deleted`!=1;");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes an array of $items and turns into a string to make a drop-down list
+ *
+ * @param $items array of items (options) to be made into a drop-down
+ * @return string of options
+ */
+function makeDropDown (array $items): string {
+    $dropDownString = "";
+    foreach ($items as $item) {
+        $dropDownString .= '<option value="' . $item['title'] . '">' . $item['title'] . '</option>';
+    }
+    return $dropDownString;
+}
+
+/**
+ * Takes selected item POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData assoc array  user response for item to select
+ * @return array of item values
+ */
+function portFolioFill (PDO $db, array $postData):array {
+    $selectedItem=$postData['itemSelect'];
+    $query=$db->prepare("SELECT `title`,`description`,`imgRef`,`projURL`,`github`,`images`.`URL`
+                               FROM `portfolioItems`
+                               LEFT JOIN `images`
+                               ON `portfolioItems`.`imgRef`
+                               =`images`.`id`
+                               WHERE `title`='$selectedItem' && `title` != 'submitPF';");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Gets list of articles from DB
+ *
+ * @param $db PDO to select from
+ * @return array of articles
+ */
+function ArticleList (PDO $db):array {
+    $query=$db->prepare("SELECT `title` FROM `articles` WHERE `deleted`!=1 ");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes selected article POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData assoc array userresponse for article to select
+ * @return array
+ */
+function SelectArt (PDO $db, array $postData):array {
+    $artSelect = $postData['artSelect'];
```

Comment:
same comment as above, just pass the data u need

Reply:
resolved